logs.go

```diff
@@ -91,18 +91,23 @@ func GetPodContainers(runner Runner, project, name string) ([]string, error) {
 // LoggableResource. Set maxLines to limit how many lines are fetched. Logs are
 // written to out and eventual error messages go to errOut.
 func FetchLogs(r Runner, resource LoggableResource, maxLines int) Task {
-	return ocLogs(r, resource, maxLines, nil, "logs")
+	return ocLogs(r, resource, maxLines, nil, "logs", false)
 }
 
 // FetchPreviousLogs is like FetchLogs, but for the previous version of a
 // resource.
 func FetchPreviousLogs(r Runner, resource LoggableResource, maxLines int) Task {
-	return ocLogs(r, resource, maxLines, []string{"--previous"}, "logs-previous")
+	return ocLogs(r, resource, maxLines, []string{"--previous"}, "logs-previous", true)
 }
 
 // ocLogs fetches logs from an OpenShift resource using oc.
-func ocLogs(r Runner, resource LoggableResource, maxLines int, extraArgs []string, what string) Task {
-	return func() error {
+func ocLogs(r Runner, resource LoggableResource, maxLines int, extraArgs []string, what string, ignoreErrors bool) Task {
+	return func() (err error) {
+		defer func() {
+			if ignoreErrors {
+				err = IgnoreError(err)
```

Comment:
Yeah, it wraps the original error with another value.

Deferred calls can mutable named return values. Not something we want to abuse though, as it gets more confusing to reason about.

Just went with it here but would also work to change the last line to add an if there.
The problem is that if you have multiple return points, then you need to remember to add that logic in every one of them -- or use the deferred call.